storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -117,5 +199,103 @@ func (s *peersSource) Read(
 
 	wg.Wait()
 
+	close(incrementalQueue)
+	incrementalWg.Wait()
+
+	if incremental {
+		// Now cache the incremental results
+		shards := make([]uint32, 0, len(shardsTimeRanges))
+		for shard := range shardsTimeRanges {
```

Comment:
Should we look at the (un)fulfilled time ranges in `result` to figure out which shards have we successfully bootstrapped instead?